sdk/supercontract_v2_transaction.go

```diff
@@ -0,0 +1,501 @@
+	if err != nil {
+		return nil, err
+	}
+
+	contractKey, err := NewAccountFromPublicKey(dto.Tx.ContractKey, atx.NetworkType)
+	if err != nil {
+		return nil, err
+	}
+
+	mosaics := make([]*MosaicId, len(dto.Tx.ServicePayments))
+
+	for i, mosaic := range dto.Tx.ServicePayments {
+		msc, err := mosaic.toStruct()
+		if err != nil {
+			return nil, err
+		}
+
+		mosaics[i] = msc
+	}
+
+	return &ManualCallTransaction{
+		*atx,
+		contractKey,
+		dto.Tx.FileNameSize,
+		dto.Tx.FunctionNameSize,
+		dto.Tx.ActualArgumentsSize,
+		dto.Tx.ExecutionCallPayment.toStruct(),
+		dto.Tx.DownloadCallPayment.toStruct(),
+		dto.Tx.ServicePaymentsCount,
+		dto.Tx.FileName,
+		dto.Tx.FunctionName,
+		dto.Tx.ActualArguments,
+		mosaics,
+	}, nil
+}
+
+// deploy contract transaction
+func NewDeployContractTransaction(
+	deadline 							*Deadline,
+	driveKey 							*PublicAccount,
+	fileNameSize 						uint16,
+	functionNameSize 					uint16,
+	actualArgumentsSize 				uint16,
+	executionCallPayment 				Amount,
+	downloadCallPayment 				Amount,
+	servicePaymentsCount 				uint8,
+	automaticExecutionFileNameSize		uint16,
+	automaticExecutionFunctionNameSize 	uint16,
+	automaticExecutionCallPayment 		Amount,
+	automaticDownloadCallPayment 		Amount,
+	automaticExecutionsNumber 			uint32,
+	assignee							*PublicAccount,
+	fileName 							string,
+	functionName 						string,
+	actualArguments 					string,
+	servicePayments 					[]*MosaicId,
+	automaticExecutionFileName			string,
+	automaticExecutionFunctionName		string,
+	networkType 			NetworkType,
+) (*DeployContractTransaction, error) {
+
+	tx := DeployContractTransaction{
+		AbstractTransaction: AbstractTransaction{
+			Deadline:    deadline,
+			Version:     DeployContractVersion,
+			Type:        DeployContract,
+			NetworkType: networkType,
+		},
+		DriveKey: 							driveKey,
+		FileNameSize: 						fileNameSize,
+		FunctionNameSize: 					functionNameSize,
+		ActualArgumentsSize: 				actualArgumentsSize,
+		ExecutionCallPayment: 				executionCallPayment,
+		DownloadCallPayment: 				downloadCallPayment,
+		ServicePaymentsCount: 				servicePaymentsCount,
+		AutomaticExecutionFileNameSize:		automaticExecutionFileNameSize,
+		AutomaticExecutionFunctionNameSize: automaticExecutionFunctionNameSize,
+		AutomaticExecutionCallPayment: 		automaticExecutionCallPayment,
+		AutomaticDownloadCallPayment:		automaticDownloadCallPayment,
+		AutomaticExecutionsNumber: 			automaticExecutionsNumber,
+		Assignee: 							assignee,
+		FileName: 							fileName,
+		FunctionName: 						functionName,
+		ActualArguments: 					actualArguments,
+		ServicePayments: 					servicePayments,
+		AutomaticExecutionFileName: 		automaticExecutionFileName,
+		AutomaticExecutionFunctionName: 	automaticExecutionFunctionName,
+	}
+
+	return &tx, nil
+}
+
+func (tx *DeployContractTransaction) GetAbstractTransaction() *AbstractTransaction {
+	return &tx.AbstractTransaction
+}
+
+func (tx *DeployContractTransaction) Bytes() ([]byte, error) {
```

Comment:
`DeployContractTransaction` and `ManualCallTransaction` have a lot of common fields. Maybe it makes sense to move them to a separate func, where these fields will be converted to vectors.

Reply:
https://github.com/proximax-storage/go-xpx-chain-sdk/commit/9fb76208c966da7be86d54c92222ac996ed3a42c
I have made some changes on this issue, is the changes ok?